xarray/core/dataset.py

```diff
@@ -7074,5 +7074,43 @@ def query(
         # apply the selection
         return self.isel(indexers, missing_dims=missing_dims)
 
+    def drop_duplicates(
+        self,
+        dims: Union[None, Hashable, Sequence[Hashable]] = None,
+        keep: Union[str, bool] = "first",
+    ):
+        """Returns a new dataset with duplicate dimension values removed.
+
+        Parameters
+        ----------
+        dims : dimension label or sequence of labels, optional
+            Only consider certain dimensions for identifying duplicates, by
+            default use all of the columns.
+        keep : {"first", "last", False}, default: "first"
+            Determines which duplicates (if any) to keep.
+            - ``first`` : Drop duplicates except for the first occurrence.
+            - ``last`` : Drop duplicates except for the last occurrence.
+            - False : Drop all duplicates.
+
+        Returns
+        -------
+        Dataset
+        """
+        if dims is None:
+            dims = list(self.coords)
+        elif isinstance(dims, str) or not isinstance(dims, Iterable):
+            dims = [dims]
+        else:
+            dims = list(dims)
+
+        new = self.copy(deep=False)
+        for dim in dims:
+            if dim not in self.dims:
+                raise ValueError("%s must be a single dataset dimension" % dim)
+            index = new.get_index(dim).duplicated(keep=keep)
+            new = new.isel(**{dim: ~index})
```

Comment:
Can this be taken out of the loop? I.e. accumulate all of them in a dict and call `isel` only once? I _think_ that's the same as calling `isel` repeatedly. 

---
Re the error you get - `dim` is `Hashable` which you cannot pass to a function. Try:

```suggestion
            new = new.isel({dim: ~index})
```